Sources/StreamChat/Database/DTOs/ChannelReadDTO.swift

```diff
@@ -85,6 +85,8 @@ extension NSManagedObjectContext {
     }
     
     func markChannelAsRead(cid: ChannelId, userId: UserId, at: Date) {
+        guard currentUser?.user.id == userId else { return }
```

Comment:
Worth adding a test to avoid regressions 👍

Reply:
Yep, still in draft state because many things are missing 😄